### DISCUSSION TITLE
Does UnrealCLR allow to incorporate existing C# libraries and frameworks into UE4?

### DISCUSSION THREAD

**Discussion by ZenaMel** [nxrighthere/UnrealCLR#69] [Help] [answered]
let's say I want to incorporate `CefSharp` into my UE4 project so that I can instantiate an interactable `web-browser` inside my game world. Would UnrealCLR lay the foundation for that?

**Comment by nxrighthere** [comment, marked as answer]
Yes, it's possible. Some libraries might not work correctly with the plugin primary because of the assembly load context. In such cases, [debugging](https://docs.microsoft.com/en-us/dotnet/standard/assembly/unloadability) is required to determine and report issues to the library maintainers, so they can enable assembly unloadability.

**Reply by nxrighthere** [reply]
For instance, [here's](https://twitter.com/nx_righthere/status/1307594908644052994) one of my tests with [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet) straight from NuGet.
